tiledb/sm/cpp_api/profile_experimental.h

```diff
@@ -0,0 +1,159 @@
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the experimental profile C++ API.
+ */
+
+#ifndef TILEDB_CPP_PROFILE_EXPERIMENTAL_H
+#define TILEDB_CPP_PROFILE_EXPERIMENTAL_H
+
+#include "context.h"
+#include "deleter.h"
+#include "exception.h"
+#include "tiledb_experimental.h"
+
+#include <memory>
+#include <string>
+
+namespace tiledb {
+
+class ProfileException : public TileDBError {
+ public:
+  explicit ProfileException(const std::string& msg)
+      : TileDBError(std::string("[TileDB::C++ API][Profile]: " + msg)) {
+  }
+};
+
+class Profile {
+ public:
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+
+  /** Default constructor. */
+  Profile()
+      : Profile(nullptr, nullptr) {
+  }
+
+  /**
+   * Constructor.
+   *
+   * @note Use of `homedir` is intended primarily for testing purposes, to
+   * preserve local files from in-test changes. Users may pass their own
+   * `homedir`, but are encouraged to use `nullptr`, the default case.
+   *
+   * @param name The profile name if you want to override the default.
+   * @param homedir The local $HOME directory path if you want to override the
+   * default.
+   */
+  explicit Profile(
+      std::optional<std::string> name = std::nullopt,
+      std::optional<std::string> homedir = std::nullopt) {
+    const char* n = name.has_value() ? name->c_str() : nullptr;
+    const char* h = homedir.has_value() ? homedir->c_str() : nullptr;
+
+    tiledb_profile_t* capi_profile;
+    tiledb_error_t* capi_error = nullptr;
+
+    int rc = tiledb_profile_alloc(n, h, &capi_profile, &capi_error);
+    if (rc != TILEDB_OK)
+      throw ProfileException(
+          "Failed to create Profile due to an internal error.");
+
+    profile_ = std::shared_ptr<tiledb_profile_t>(capi_profile, Profile::free);
+  }
+
+  /** Copy and move constructors are deleted. Profile objects are immutable. */
+  Profile(const Profile&) = delete;
+  Profile(Profile&&) = delete;
+  Profile& operator=(const Profile&) = delete;
+  Profile& operator=(Profile&&) = delete;
+
+  /** Destructor. */
+  ~Profile() = default;
+
+  /* ********************************* */
+  /*                API                */
+  /* ********************************* */
+
+  /** Returns the C TileDB profile object. */
+  std::shared_ptr<tiledb_profile_t> ptr() const {
+    return profile_;
+  }
+
+  /** Retrieves the name of the profile. */
+  std::string get_name() const {
+    tiledb_error_t* capi_error = nullptr;
+    tiledb_string_t* name;
+
+    int rc = tiledb_profile_get_name(profile_.get(), &name, &capi_error);
+    if (rc != TILEDB_OK)
+      throw ProfileException(
+          "Failed to retrieve profile name due to an internal error.");
+
+    const char* out_ptr;
+    size_t out_length;
+    tiledb_string_view(name, &out_ptr, &out_length);
+    std::string out_str(out_ptr, out_length);
```

Comment:
Should this return `std::string_view` instead? Seems like a plus to avoid the `std::string` construction.

Reply:
For posterity, consensus was: we will return `std::string` because of the intermediate `tiledb_string_view` handle which must be freed.